README.md

```diff
@@ -53,34 +73,70 @@ let document = "# My Great Document \
 
 let root = parse_document(document);
 
-let renderer = root.capabilities.render.as_ref().unwrap();
-println!("{}", renderer.render_xml());
+
+println!("{}", root.render_xml());
 ```
 
-### API
+### Examples
 
-The API is organized around capability objects attached to Nodes. Specifically
+* Transform all text into uppercase
+    ```
+    use doogie::parse_document;
+
+    let document = "# My Great Document \
+    \
+    * Item 1 \
+    * Item 2 \
+    * Item 3";
+
+    let root = parse_document(document);
+
+    for (node, _) in root.iter() {
+        if let Node::Text(ref node) = node {
+            let content = node.get_content().unwrap();
+            node.set_content(&content.to_uppercase()).unwrap();
+        }
+    }
+
+    ```
+* Remove all level 6 `Heading` nodes
+    ```
+    use doogie::parse_document;
+
+    let document = "# My Great Document \
+    \
+    * Item 1 \
+    * Item 2 \
+    * Item 3";
+
+    let root = parse_document(document);
+
+    for (node, _) in root.iter() {
+        if let Node::Heading(ref heading) = node {
+            if heading.get_level() == 6 {
+                heading.unlink();
```

Comment:
I'd expect something side affecting like `unlink` to require a `&mut self`

Reply:
That's an interesting question. Nothing in Rust-land requires that the function take a mutable self, so it would be sort of like artificially imposing a mutable function signature as a way of signalling intent. The compiler would complain with a warning if I did that though. Unless I factored out the interior mutability pattern used in the Resource manager. But that's just the case for link and unlink. Other functions that change the data structure such as setters would still not be able to be mutable without warnings.

Reply:
Code updated to require mutable self in setter and mutator functions.